models/xception/xception_impl.hpp

```diff
@@ -0,0 +1,227 @@
+/**
+ * @file xception_impl.hpp
+ * @author Shubham Agrawal
+ *
+ * Implementation of Xception using mlpack.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MODELS_MODELS_XCEPTION_XCEPTION_IMPL_HPP
+#define MODELS_MODELS_XCEPTION_XCEPTION_IMPL_HPP
+
+#include "xception.hpp"
+
+namespace mlpack {
+namespace models {
+
+template<typename MatType>
+XceptionType<MatType>::XceptionType(
+    const size_t numClasses,
+    const bool includeTop) :
+    ann::MultiLayer<MatType>(),
+    numClasses(numClasses),
+    includeTop(includeTop)
+{
+  MakeModel();
+}
+
+template<typename MatType>
+XceptionType<MatType>::XceptionType(
+    const XceptionType& other) :
+    ann::MultiLayer<MatType>(other),
+    numClasses(other.numClasses),
+    includeTop(other.includeTop)
+{
+  // Nothing to do here.
+}
+
+template<typename MatType>
+XceptionType<MatType>::XceptionType(
+    XceptionType&& other) :
+    ann::MultiLayer<MatType>(std::move(other)),
+    numClasses(std::move(other.numClasses)),
+    includeTop(std::move(other.includeTop))
+{
+  // Nothing to do here.
+}
+
+template<typename MatType>
+XceptionType<MatType>&
+XceptionType<MatType>::operator=(const XceptionType& other)
+{
+  if (this != &other)
+  {
+    ann::MultiLayer<MatType>::operator=(other);
+    numClasses = other.numClasses;
+    includeTop = other.includeTop;
+  }
+
+  return *this;
+}
+
+template<typename MatType>
+XceptionType<MatType>&
+XceptionType<MatType>::operator=(XceptionType&& other)
+{
+  if (this != &other)
+  {
+    ann::MultiLayer<MatType>::operator=(std::move(other));
+    numClasses = std::move(other.numClasses);
+    includeTop = std::move(other.includeTop);
+  }
+
+  return *this;
+}
+
+template<typename MatType>
+template<typename Archive>
+void XceptionType<MatType>::serialize(
+    Archive& ar, const uint32_t /* version */)
+{
+  ar(cereal::base_class<ann::MultiLayer<MatType>>(this));
+
+  ar(CEREAL_NVP(numClasses));
+  ar(CEREAL_NVP(includeTop));
+}
+
+template<typename MatType>
+void XceptionType<MatType>::SeparableConv(
+    ann::MultiLayer<MatType>* block,
+    const size_t inMaps,
+    const size_t outMaps,
+    const size_t kernelSize,
+    const size_t stride = 1,
+    const size_t padding = 0,
+    const bool useBias = false)
+{
+  block->template Add<ann::GroupedConvolution>(inMaps, kernelSize, kernelSize,
+    inMaps, stride, stride, padding, padding, "none", useBias);
+  block->template Add<ann::Convolution>(outMaps, 1, 1, 1, 1, 0, 0, "none",
+    useBias);
```

Comment:
```suggestion
      inMaps, stride, stride, padding, padding, "none", useBias);
  block->template Add<ann::Convolution>(outMaps, 1, 1, 1, 1, 0, 0, "none",
      useBias);
```
Tiny style fixes, wrapped lines should be indented twice. :+1: